src/flows/recommendation_api/user_impressions_flow.py

```diff
@@ -1,49 +1,84 @@
-from prefect import task, Flow, Parameter, unmapped
-from prefect.tasks.aws.s3 import S3List
+from prefect import task, Flow, Parameter, context
+from api_clients.pocket_snowflake_query import PocketSnowflakeQuery
 import pandas as pd
+import datetime
 import boto3
 from sagemaker.feature_store.feature_group import FeatureGroup
 from sagemaker.session import Session
+from prefect.schedules import IntervalSchedule
 
+from utils import config
 from utils.flow import get_flow_name
 
 FLOW_NAME = get_flow_name(__file__)
 
-
-@task
-def extract_parquet_as_df(bucket:str, parquet_file: str):
-    parquet_file = f"s3://{bucket}/{parquet_file}"
-    return pd.read_parquet(parquet_file)
+BASE_QUERY = """
+WITH 
+prep as (
+  SELECT
+    a.HASHED_USER_ID,
+    a.CONTENT_ID,
+    c.RESOLVED_ID,
+    MAX(a.IMPRESSION_AGE) as max_impression_age,
+    SUM(a.IMPRESSION_COUNT) as total_imprs,
+    MAX(a.HAPPENED_AT_DAY) as HAPPENED_AT
+  FROM ANALYTICS.DBT.FIRST_IMPRESSED_AGE as a
+  JOIN ANALYTICS.DBT.CONTENT AS c
+    ON c.CONTENT_ID = a.CONTENT_ID
+  WHERE CURRENT_DATE - DATE(a.TIME_ADDED) <= 1  
+  GROUP BY 1, 2, 3
+  )
+  
+SELECT
+    u.USER_ID,
+    p.HAPPENED_AT AS UPDATED_AT,
+    LISTAGG(DISTINCT p.RESOLVED_ID, ',') AS RESOLVED_IDS
```

Comment:
```suggestion
    ARRAY_AGG(DISTINCT p.RESOLVED_ID) AS RESOLVED_IDS
```
[array_agg](https://docs.snowflake.com/en/sql-reference/functions/array_agg.html) surrounds the values by brackets.

Reply:
nice

Reply:
this inserts newline characters to make an array it seems: `[id1\n, id2\n, ..., id33\n]` .  i'm going back to listagg